pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -356,6 +356,96 @@ final class Event {
+    String? userPlatform,
+    String? devtoolsPlatform,
+    String? devtoolsChrome,
+    String? devtoolsVersion,
+    String? ideLaunched,
+    String? isExternalBuild,
+    String? isEmbedded,
+    String? ideLaunchedFeature,
+
+    // PerformanceScreenMetrics
+    int? uiDurationMicros,
+    int? rasterDurationMicros,
+    int? shaderCompilationDurationMicros,
+    int? traceEventCount,
+
+    // ProfilerScreenMetrics
+    int? cpuSampleCount,
+    int? cpuStackDepth,
+
+    // MemoryScreenMetrics
+    int? heapDiffObjectsBefore,
+    int? heapDiffObjectsAfter,
+    int? heapObjectsTotal,
+
+    // InspectorScreenMetrics
+    int? rootSetCount,
+    int? rowCount,
+    int? inspectorTreeControllerId,
+  })  : eventName = DashEvent.devtoolsEvent,
+        eventData = {
+          'eventCategory': eventCategory,
+          'label': label,
+          'value': value,
+
+          'userInitiatedInteraction': userInitiatedInteraction,
+
+          // Optional parameters
+          if (g3Username != null) 'g3Username': g3Username,
+          if (userApp != null) 'userApp': userApp,
+          if (userBuild != null) 'userBuild': userBuild,
+          if (userPlatform != null) 'userPlatform': userPlatform,
+          if (devtoolsPlatform != null) 'devtoolsPlatform': devtoolsPlatform,
+          if (devtoolsChrome != null) 'devtoolsChrome': devtoolsChrome,
+          if (devtoolsVersion != null) 'devtoolsVersion': devtoolsVersion,
+          if (ideLaunched != null) 'ideLaunched': ideLaunched,
+          if (isExternalBuild != null) 'isExternalBuild': isExternalBuild,
+          if (isEmbedded != null) 'isEmbedded': isEmbedded,
+          if (ideLaunchedFeature != null)
+            'ideLaunchedFeature': ideLaunchedFeature,
+
+          // PerformanceScreenMetrics
+          if (uiDurationMicros != null) 'uiDurationMicros': uiDurationMicros,
+          if (rasterDurationMicros != null)
+            'rasterDurationMicros': rasterDurationMicros,
+          if (shaderCompilationDurationMicros != null)
+            'shaderCompilationDurationMicros': shaderCompilationDurationMicros,
+          if (traceEventCount != null) 'traceEventCount': traceEventCount,
+
+          // ProfilerScreenMetrics
+          if (cpuSampleCount != null) 'cpuSampleCount': cpuSampleCount,
+          if (cpuStackDepth != null) 'cpuStackDepth': cpuStackDepth,
+
+          // MemoryScreenMetrics
+          if (heapDiffObjectsBefore != null)
+            'heapDiffObjectsBefore': heapDiffObjectsBefore,
+          if (heapDiffObjectsAfter != null)
+            'heapDiffObjectsAfter': heapDiffObjectsAfter,
+          if (heapObjectsTotal != null) 'heapObjectsTotal': heapObjectsTotal,
+
+          // InspectorScreenMetrics
+          if (rootSetCount != null) 'rootSetCount': rootSetCount,
+          if (rowCount != null) 'rowCount': rowCount,
+          if (inspectorTreeControllerId != null)
+            'inspectorTreeControllerId': inspectorTreeControllerId,
+        };
+
```

Comment:
Does this look like everything is being captured from the `_gtagEvent` function?

Referenced from here: https://github.com/eliasyishak/devtools/blob/migrate-select-event/packages/devtools_app/lib/src/shared/analytics/_analytics_web.dart#L196-L254

Reply:
We have included all of the fields from the linked function above, except for `sent_to` and `flutter_client_id` and that is because we have the "send to" for GA4 within the package and the client id is managed by the package and added to the payload on send requests

Reply:
cc @kenzieschmoll ^